pytorch_lightning/utilities/cli.py

```diff
@@ -112,8 +116,9 @@ def __init__(
         self,
         model_class: Type[LightningModule],
         datamodule_class: Type[LightningDataModule] = None,
-        save_config_callback: Type[SaveConfigCallback] = SaveConfigCallback,
+        save_config_callback: Optional[Type[SaveConfigCallback]] = SaveConfigCallback,
         save_config_filename: str = 'config.yaml',
+        save_config_overwrite: bool = False,
```

Comment:
Adding `save_config_overwrite` could make sense for some use cases. However I don't think this is the solution for #8003. There shouldn't be a race condition. #8003 should be solved by #8017. `SaveConfigCallback` should only attempt to save the config for rank 0.